unlock_indicator.c

```diff
@@ -91,6 +94,97 @@ static double scaling_factor(void) {
     return (dpi / 96.0);
 }
 
+/*
+ * Returns the cairo_font_face_t for sans-serif
+ * if this thing returns NULL, then we're somehow on a system without a
+ * sans-serif font, or a system without fontconfig (but with cairo somehow)
+ * Reference counts and caches the font-face, so we don't have memory leaks
+ * (and so that we don't have to go through all the parsing each time)
+ */
+
+static cairo_font_face_t *get_font_face() {
+    if (cairo_font_face_get_reference_count(sans_serif)) {
+        return cairo_font_face_reference(sans_serif);
+    }
+
+    FcResult result;
+    /*
+     * Loads the default config.
+     * On successive calls, does no work and just returns true.
+     */
+    if (!FcInit()) {
+        DEBUG("Fontconfig init failed. No fonts will be available...\n");
+        return NULL;
+    }
+    /*
+     * converts a font face name to a pattern for that face name
+     */
+    FcPattern *pattern = FcNameParse(font_face_name);
+    if (!pattern) {
+        DEBUG("no sans-serif font available\n");
+        return NULL;
+    }
+
+    /*
+     * Gets the default font for our pattern. (Gets the default sans-serif font face)
+     * Required to make the following FcFontMatch call for some FontConfig bookkeeping.
+     */
+    FcDefaultSubstitute(pattern);
+    if (!FcConfigSubstitute(FcConfigGetCurrent(), pattern, FcMatchPattern)) {
+        DEBUG("config sub failed?\n");
+        return NULL;
+    }
+    /*
+     * Looks up the font pattern and does some internal RenderPrepare work,
+     * then returns the result.
+     */
+    FcPattern *pattern_ready = FcFontMatch(FcConfigGetCurrent(), pattern, &result);
```

Comment:
result is assigned but not used. Can we either check it or remove it?

Reply:
This is a fun one. It has to be assigned (or else fontconfig will error out), but its value on a success isn't well-defined. In [fonctconfig's provided examples](https://github.com/behdad/fontconfig/blob/7ca28c2fedb34c1db5ee3116d335f15195859db0/fc-match/fc-match.c#L213) they also only check to see if they get back `NULL` or not :(

I'll update the commenting to say `returns the resulting pattern` to be a bit more clear, though.